src/Shared/PackageManagers/MavenProjectManager.cs

```diff
@@ -285,13 +416,25 @@ public override async Task<bool> PackageVersionExistsAsync(PackageURL purl, bool
             return metadata;
         }
 
-        public override Uri GetPackageAbsoluteUri(PackageURL purl)
+        public override Uri? GetPackageAbsoluteUri(PackageURL purl)
         {
-            string? packageNamespace = purl?.Namespace?.Replace('.', '/');
             string? packageName = purl?.Name;
-            string feedUrl = (purl?.Qualifiers?["repository_url"] ?? ENV_MAVEN_ENDPOINT).EnsureTrailingSlash();
+            string feedUrl = purl?.Qualifiers?["repository_url"] ?? ENV_MAVEN_ENDPOINT.GetRepositoryUrl();
+            MavenSupportedUpstream upstream = feedUrl.GetMavenSupportedUpstream();
+
+            if (upstream == MavenSupportedUpstream.MavenCentralRepository)
+            {
+                string? packageNamespace = purl?.Namespace?.Replace('.', '/');
+                return new Uri($"{upstream.GetRepositoryUrl().EnsureTrailingSlash()}{packageNamespace}/{packageName}");
+            }
+            else if (upstream == MavenSupportedUpstream.GoogleMavenRepository)
+            {
+                string packageNamespace = purl.Namespace;
+                return new Uri($"{upstream.GetRepositoryUrl()}{packageNamespace}/{packageName}");
+            }
```

Comment:
Is there a way we could have a single implementation for this? The Google Maven repository must follow the same directory structure right?

I forget the intent for this particular method though - is this method intended to provide a URL that users can click on or something?

Reply:
The documentation for this method is: "Get the uri for the package home page (no version)". So it should yield something like: https://maven.google.com/web/index.html#android.arch.core:common (vs Maven Central: https://repo1.maven.org/maven2/ant/ant/).

Reply:
The only place I can find it being used is to build a SARIF Result.Location object for a given package URL (I think for logging purposes?). It doesn't seem that critical; we could also leave this method unimplemented.

Reply:
I removed this method. Feel free to reopen this thread if you have more thoughts.